src/apps/DIFFUSION3DPA-Cuda.cpp

```diff
@@ -0,0 +1,292 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-21, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "DIFFUSION3DPA.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_CUDA)
+
+#include "common/CudaDataUtils.hpp"
+
+#include <iostream>
+
+namespace rajaperf {
+namespace apps {
+
+#define DIFFUSION3DPA_DATA_SETUP_CUDA                                       \
+  allocAndInitCudaDeviceData(Basis, m_B, DPA_Q1D *DPA_D1D);                 \
+  allocAndInitCudaDeviceData(dBasis, m_G, DPA_Q1D *DPA_D1D);                \
+  allocAndInitCudaDeviceData(D, m_D, DPA_Q1D *DPA_Q1D *DPA_Q1D *SYM *m_NE); \
+  allocAndInitCudaDeviceData(X, m_X, DPA_D1D *DPA_D1D *DPA_D1D *m_NE);      \
+  allocAndInitCudaDeviceData(Y, m_Y, DPA_D1D *DPA_D1D *DPA_D1D *m_NE);
+
+#define DIFFUSION3DPA_DATA_TEARDOWN_CUDA                            \
+  getCudaDeviceData(m_Y, Y, DPA_D1D *DPA_D1D *DPA_D1D *m_NE);       \
+  deallocCudaDeviceData(Basis);                                     \
+  deallocCudaDeviceData(dBasis);                                    \
+  deallocCudaDeviceData(D);                                         \
+  deallocCudaDeviceData(X);                                         \
+  deallocCudaDeviceData(Y);
+
+//#define USE_RAJA_UNROLL
```

Comment:
These macro definitions are repeated in other kernel files (MASS3DPA) and in this PR. Let's move them into one header file and include that where we need them.